flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/BinaryWriter.java

```diff
@@ -68,7 +69,7 @@
 
 	void writeRow(int pos, BaseRow value, BaseRowSerializer type);
 
-	void writeGeneric(int pos, BinaryGeneric value);
+	void writeGeneric(int pos, BinaryGeneric value, BinaryGenericSerializer innerSerializer);
```

Comment:
Or just use real inner serializer instead of `BinaryGenericSerializer`?

Reply:
I considered that. The thing is that from the interface perspective imho it makes more sense to use `BinaryGenericSerializer` here. In the end we are serializing the `BinaryGeneric` the inner serializer is an implementation detail.

Moreover this method is called from couple of places, e.g. `CodeGenUtils#binaryWriterWriteField` where it transitively calls `InternalSerializers#create`. I am pretty sure it should return the `BinaryGenericSerializer` rather than the internal one.

Generally speaking I think it is safer to assume the BinaryGenericSerializer in this method. This way we limit the scope of extracting the inner serializer to a single method.